### PR TITLE
Feat/260 frontend fetch wrapper and auth service

### PR DESCRIPTION
**Pull request by ianliuwk1019**
Part 1 of #260:
* (This one is removed in favor of Axios). Add a Fetch http call wrapper: for adding auth header and intercept request/response if needed.
* Install and use Axios for http calls and use it first on Get Applications endpoint call with simple request interceptor added.
* Add an AuthService: Contains a readyOnly User(token) state with localstorage, and convenient login/logout methods.
* Header menu show/hide based on login/logout (temporarily with fake token).
* A Cognito env is added (probably removed later when impl is clear in future)
* This does not include Auth with Cognito yet (second part).

### CODE REVIEW

File: frontend/src/services/ApiService.ts [hunk 1]
```diff
@@ -1,18 +1,21 @@
...
-    private apiUrl:string
+    private _apiUrl:string
 
     constructor() {
         const environmentSettings = new EnvironmentSettings()
-        this.apiUrl = environmentSettings.getApiBaseUrl() + '/api/v1'
+        this._apiUrl = environmentSettings.getApiBaseUrl() + '/api/v1'
     }
 
+    public get apiUrl(): string {
```
Comment (basilv) [inline frontend/src/services/ApiService.ts:13]: Not sure why need a public getter for the Url

File: frontend/src/services/ApiService.ts [hunk 2]
```diff
@@ -1,18 +1,21 @@
+import type { Application } from '@/services/ApplicationState';
 import { EnvironmentSettings } from '@/services/EnvironmentSettings';
-import type { Application } from '@/services/ApplicationState'
 
 export class ApiService {
 
-    private apiUrl:string
+    private _apiUrl:string
```
Comment (basilv) [inline frontend/src/services/ApiService.ts:6]: Is this a typescript convention? (Prefix private with underscore?)

File: frontend/src/services/AuthService.ts [hunk 3]
```diff
@@ -0,0 +1,56 @@
+import router from '@/router';
+import { reactive, readonly } from 'vue';
+
+// Auth state from localStorage
+const state = reactive({
```
Comment (basilv) [inline frontend/src/services/AuthService.ts:5]: You don't necessarily need to use local storage for the logged in user versus just using Vue's global state. 
(I know we used local storage in FOM).

File: frontend/src/services/AuthService.ts [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+        name: 'fake_user',
+        token: 'fake_token'
+    };
+
+    // update famUser state
+    state.famUser = famUser;
+
+    // store user details and jwt in local storage to keep user logged in between page refreshes
+    localStorage.setItem('famUser', JSON.stringify(famUser));
+    router.push('/');
```
Comment (basilv) [inline frontend/src/services/AuthService.ts:31]: Not sure why redirect to home page? What if user is in the middle of a page and login times out? (We probably haven't worried about that flow yet, might be post MVP.)

File: frontend/src/services/AuthService.ts [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+}
+
+async function logout() {
+    state.famUser = null;
+    localStorage.removeItem('famUser');
```
Comment (basilv) [inline frontend/src/services/AuthService.ts:36]: For vue ref(), should use state.value.famUser... Not sure about the reactive() variant.

File: frontend/src/services/AuthService.ts [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+    // TODO: Probably need to call Amplify library for sign out from Cognito?
+    router.push('/');
+}
+
+// -----
```
Comment (basilv) [inline frontend/src/services/AuthService.ts:41]: This section below confuses me, not sure what you are trying to do with this.

**Review by ConradBoydElliottGustafson — COMMENTED** [review]
I don't really know the frontend code well enough to review. Leaving it for @basilv

**Review by basilv — APPROVED** [review]

**Pull request merged**